mantidproject/mantid
# Fix 0 / False vals being defaulted isis sans

**Description of work.**

    Prevent falsey types being reset back to default in ISIS SANS
    
    Falsey types, such as False or 0.0 were being picked up in multiple
    layers as an empty entry. This was causing users to have weird effects,
    such as setting a Phi Limit from 0-90 defaulting back to -90,90 or tick
    boxes reticking themselves.
    
    A helper was added for the model layers to unify this behaviour of what
    is an acceptable falsey type.
    
    The presenter layer was updated to only filter out empty strings and
    none types, so that Qt doesn't choke on them.
    
    The view was updated to remove logic (which shouldn't be in the view)
    that was doing a final check before setting the value.



**Report to:** ISIS/Sarah (SANS) - Done

**To test:**
- Download the ISIS Training data set
- Open Mantid and SANS->ISIS SANS
- Edit the MaskFile.txt and add `L/PHI/NOMIRROR 0 180` anywhere
- Point the user file to the .txt file called MaskFile in LoqDemo
- Open the CSV file in the same folder
- Go back to the SANS GUI, then settings.
- Go to the Q, Wavelength tab and check that the Phi Limit is 0-180 and that `Use Mirror` is unticked


Fixes #30217 .





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
